stl/inc/iterator

```diff
@@ -757,48 +731,45 @@ public:
 
     using _Prevent_inheriting_unwrap = counted_iterator;
 
-    // clang-format off
-    constexpr void _Verify_offset(const iter_difference_t<_Iter> _Off) const
-        requires _Offset_verifiable_v<_Iter>
-    {
+    constexpr void _Verify_offset(const iter_difference_t<_Iter> _Off) const requires _Offset_verifiable_v<_Iter> {
         _STL_VERIFY(_Off < _Length, "Offset larger than counted_iterator length");
         _Current._Verify_offset(_Off);
     }
 
-    _NODISCARD constexpr counted_iterator<_Unwrapped_t<const _Iter&>> _Unwrapped() const&
-        requires _Unwrappable_v<const _Iter&>
-    {
+    _NODISCARD constexpr counted_iterator<_Unwrapped_t<const _Iter&>>
+        _Unwrapped() const& requires _Unwrappable_v<const _Iter&> {
         return static_cast<counted_iterator<_Unwrapped_t<const _Iter&>>>(_Current._Unwrapped());
     }
 
-    _NODISCARD constexpr counted_iterator<_Unwrapped_t<_Iter>> _Unwrapped() &&
-        requires _Unwrappable_v<const _Iter&>
-    {
+    _NODISCARD constexpr counted_iterator<_Unwrapped_t<_Iter>> _Unwrapped() && requires _Unwrappable_v<const _Iter&> {

```

Comment:
I do not really like this as it is difficult to discern the `rvlaue qualifier` from the `and` of a constraint-expression

Reply:
C++11 did programmers a disservice by adding "rvalue reference" and "forwarding reference" to the pre-C++11 meaning of `&&`. The best support I have here is that anything that precedes the first `requires` in the declaration is not part of a constraint-expression.

I wonder if we could convince the other STL maintainers to use `and` and `or` for the logical operations and reserve `&&` for rvalue/forwarding references including rvalue qualifiers.

Reply:
Remember that MSVC doesn't accept the "alternative representations" in permissive mode.